### PR TITLE
lib: add m0_bitmap_ffs()

### PR DESCRIPTION
**Pull request by andriytk**
Add "find fist bit set" function to lib/bitmap.

### CODE REVIEW

File: lib/bitmap.h [hunk 1]
```diff
@@ -86,6 +86,14 @@ M0_INTERNAL void m0_bitmap_fini(struct m0_bitmap *map);
  */
 M0_INTERNAL bool m0_bitmap_get(const struct m0_bitmap *map, size_t idx);
 
+/**
+   Find first non-zero (a.k.a. set, true) bit from a bitmap.
```
Comment (nikitadanilov) [inline lib/bitmap.h:90]: "Finds the".

File: lib/bitmap.c [hunk 2]
```diff
@@ -109,6 +109,20 @@ M0_INTERNAL bool m0_bitmap_get(const struct m0_bitmap *map, size_t idx)
...
+{
+	size_t i;
+	size_t idx;
+
+	for (i = 0; i < M0_BITMAP_WORDS(map->b_nr); i++) {
+		idx = __builtin_ffsll(map->b_words[i]);
+		if (idx > 0)
+			return i * M0_BITMAP_BITS + (idx - 1);
+	}
+	return (size_t)-1;
```
Comment (nikitadanilov) [inline lib/bitmap.c]: Why the cast is needed?

**Review by nikitadanilov — CHANGES_REQUESTED** [review]

File: lib/bitmap.c (hunk 2 above)
Comment (andriytk) [reply lib/bitmap.c]: Fixed.

File: lib/bitmap.h (hunk 1 above)
Comment (andriytk) [reply lib/bitmap.h]: Fixed.

File: lib/bitmap.c [hunk 3]
```diff
@@ -109,16 +109,30 @@ M0_INTERNAL bool m0_bitmap_get(const struct m0_bitmap *map, size_t idx)
...
 
-M0_INTERNAL size_t m0_bitmap_ffz(const struct m0_bitmap *map)
+M0_INTERNAL int m0_bitmap_ffs(const struct m0_bitmap *map)
 {
-	size_t idx;
+	int i;
+	int idx;
+
+	for (i = 0; i < M0_BITMAP_WORDS(map->b_nr); i++) {
+		idx = __builtin_ffsll(map->b_words[i]);
```
Comment (huanghua78) [inline lib/bitmap.c:118]: Is this GCC built-in function?
If we want to port to other platform, e.g. ARM, is it still available?

**Review by huanghua78 — APPROVED** [review]

**Review by max-seagate — APPROVED** [review]
[* suggestion *]

s/

> lib: add m0_bitmap_ffs()

/

lib/bitmap: add m0_bitmap_ffs()

/

In this case a bit of time is saved when someone goes through the commit history in `git log` or `git log --oneline`.

**Review by ivan-alekhin — APPROVED** [review]

**Review by mehjoshi — APPROVED** [review]

**Review by nikitadanilov — APPROVED** [review]

**Pull request merged**